native-package/src/index.js

```diff
@@ -42,34 +42,38 @@ registerNativeHandlers({
       });
     },
   },
-  pickImage: () =>
-    new Promise((resolve, reject) => {
-      ImagePicker.showImagePicker({}, (response) => {
-        if (response.error) {
-          reject(Error(response.error));
-        }
-        let { uri } = response;
-        if (Platform.OS === 'android') {
-          uri = 'file://' + response.path;
-        }
-
-        resolve({
-          cancelled: response.didCancel,
-          uri,
-        });
-      });
-    }),
   pickDocument: async () => {
     try {
-      const res = await DocumentPicker.pick({
+      const res = await DocumentPicker.pickMultiple({
         type: [DocumentPicker.types.allFiles],
       });
-      const { uri } = res;
 
       return {
         cancelled: false,
-        uri,
-        name: res.name,
+        docs: res.map(({ name, uri }) => ({ name, uri })),
+      };
+    } catch (err) {
+      return {
+        cancelled: true,
+      };
+    }
+  },
+  pickImage: async () => {
```

Comment:
We accept `maxNumberOfFiles` prop on MessageInput component. Can we add a param to this function to configure limit?

Reply:
One more thought. We can introduce some prop on MessageInput which enables (or configures) compression on image uploads and pass that prop as argument to this function. Currently we ask users to do it on their end, but instead we can just provide it OOTB (by just adding some prop)

Reply:
sure